src/main/java/com/tutorhub/web/controller/ProgressController.java

```diff
@@ -0,0 +1,24 @@
+package com.tutorhub.web.controller;
+
+import com.tutorhub.service.ProgressService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/progress")
+public class ProgressController {
+  private final ProgressService progressService;
+
+  @PostMapping
+  public void create(
+      @RequestParam final Long userId,
+      @RequestParam final Long courseId,
+      @RequestParam final Long moduleId,
+      @RequestParam final Long submoduleId) {
```

Comment:
It would be better to have some `@PathVariable` arguments, it is more common with POST requests and more obvious than 4 params

Reply:
I moved these 4 parameters into the request body

Reply:
I approved, you can merge